mongocrypt/src/convert.rs

```diff
@@ -13,15 +15,20 @@ pub(crate) fn doc_binary(doc: &Document) -> Result<BinaryBuf> {
     Ok(BinaryBuf::new(bytes))
 }
 
+pub(crate) fn path_cstring(path: &std::path::Path) -> Result<CString> {
+    let bytes = path_bytes(path)?;
+    CString::new(bytes).map_err(|e| encoding!("could not convert path to cstring: {:?}", e))
```

Comment:
This does the same "append `'\0'`" that the callers were doing manually before, and also validates that there are no internal `nul`s.